test/passes/memory-packing_all-features.wast

```diff
@@ -1,38 +1,472 @@
+    (memory.init 12
+      (i32.const 0)
+      (i32.const 135)
+      (i32.const 0)
+    )
+    (data.drop 12)
+  )
+
+  (data passive "\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00even\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00more\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00zeroes\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00") ;; 13
+
+  (func $out-of-bounds-size
+    (memory.init 13
+      (i32.const 0)
+      (i32.const 0)
+      (i32.const 135)
+    )
+    (data.drop 13)
+  )
+
+  (data passive "\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00even\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00more\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00zeroes\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00") ;; 14
+
+  (func $zero-size-at-bounds-offset
+    (memory.init 14
+      (i32.const 0)
+      (i32.const 134)
+      (i32.const 0)
+    )
+    (data.drop 14)
+  )
 )
+
 (module
- (memory $0 1 1)
- (data (i32.const 0) "")
- (func $foo
-  (memory.init 0
-   (i32.const 0)
-   (i32.const 0)
-   (i32.const 0)
-  )
- )
- (func $bar
-  (drop
-   (loop (result i32)
+  (import "env" "memory" (memory $0 2048 2048))
+  (data passive "hi\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00even\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00hi") ;; 0
+
+  (func $zero-length-init-zeroes
+    (memory.init 0
+      (i32.const 13)
+      (i32.const 10)
+      (i32.const 0)
+    )
     (data.drop 0)
-    (i32.const 42)
-   )
   )
- )
+
+  (data passive "hi\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00even\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00hi") ;; 1
+
+  (func $zero-length-init-nonzeroes
+    (memory.init 1
+      (i32.const 13)
+      (i32.const 33)
+      (i32.const 0)
+    )
+    (data.drop 1)
+  )
+
+  (data passive "\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00even\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00hi") ;; 2
+
+  (func $zero-length-init-zeroes-2
+    (memory.init 2
+      (i32.const 13)
+      (i32.const 10)
+      (i32.const 0)
+    )
+    (data.drop 2)
+  )
+
+  (data passive "\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00even\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00hi") ;; 3
+
+  (func $zero-length-init-nonzeroes-2
+    (memory.init 3
+      (i32.const 13)
+      (i32.const 31)
+      (i32.const 0)
+    )
+    (data.drop 3)
+  )
```

Comment:
What does this module test? All four test results look the same so not sure how they are different.

Reply:
This module was more relevant before I added the special handling for zero-length segments. It tests that zero-length segments are handled correctly no matter where the offset falls and what the segment looks like.